File: app/src/main/java/com/example/seg4105_calorietrackerapp/HomeActivity.java
package com.example.seg4105_calorietrackerapp;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class HomeActivity extends AppCompatActivity {

    private Button btnLogMeal;
    private ImageButton navHome, navCalendar, navSettings, btnSearch, btnProfile;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle("Homepage - Meal App");
        }

        btnLogMeal   = findViewById(R.id.btnLogMeal);
        navHome      = findViewById(R.id.navHome);
        navCalendar  = findViewById(R.id.navCalendar);
        navSettings  = findViewById(R.id.navSettings);
        btnSearch    = findViewById(R.id.btnSearch);
        btnProfile   = findViewById(R.id.btnProfile);

        // Log Meal -> ouvrir ton écran de prise de photo par ex.
        btnLogMeal.setOnClickListener(v -> {
            // Exemple : aller vers TakePhotoActivity
            // Intent intent = new Intent(this, TakePhotoActivity.class);
            // startActivity(intent);
            Toast.makeText(this, "Log meal tapped", Toast.LENGTH_SHORT).show();
        });

        navHome.setOnClickListener(v ->
                Toast.makeText(this, "Already on Home", Toast.LENGTH_SHORT).show()
        );

        navCalendar.setOnClickListener(v ->
                Toast.makeText(this, "Open calendar (TODO)", Toast.LENGTH_SHORT).show()
        );

        navSettings.setOnClickListener(v ->
                Toast.makeText(this, "Open settings (TODO)", Toast.LENGTH_SHORT).show()
        );

        btnSearch.setOnClickListener(v ->
                Toast.makeText(this, "Search tapped (TODO)", Toast.LENGTH_SHORT).show()
        );

        btnProfile.setOnClickListener(v ->
                Toast.makeText(this, "Profile tapped (TODO)", Toast.LENGTH_SHORT).show()
        );
    }
}

File: app/src/main/java/com/example/seg4105_calorietrackerapp/HomeActivity.kt
package com.example.seg4105_calorietrackerapp

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class HomeActivity : AppCompatActivity() {

    // UI references for the three summary cards
    private lateinit var tvMealsLoggedValue: TextView
    private lateinit var tvCaloriesValue: TextView
    private lateinit var tvBreakfastSummary: TextView
    private lateinit var tvLunchSummary: TextView
    private lateinit var tvDinnerSummary: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // Inflate the Home layout
        setContentView(R.layout.activity_home)

        // Setup navigation bars
        NavHelper.setupBottomNav(this, NavHelper.NavItem.HOME)
        NavHelper.setupTopNav(this)

        // Bind views from XML
        tvMealsLoggedValue = findViewById(R.id.tvMealsLoggedValue)
        tvCaloriesValue = findViewById(R.id.tvCaloriesValue)
        tvBreakfastSummary = findViewById(R.id.tvBreakfastSummary)
        tvLunchSummary = findViewById(R.id.tvLunchSummary)
        tvDinnerSummary = findViewById(R.id.tvDinnerSummary)

        // "Log Meal" opens DayView so user can add breakfast/lunch/dinner
        val btnLogMeal = findViewById<Button>(R.id.btnLogMeal)
        btnLogMeal.setOnClickListener {
            startActivity(Intent(this, DayView::class.java))
        }

        // Initial render when Home first loads
        renderFromStore()
    }

    override fun onResume() {
        super.onResume()
        renderFromStore()
    }

    private fun renderFromStore() {
        val count = MealStore.mealsLoggedCount()
        val totalCals = MealStore.totalCalories()

        // Top cards: progress + total calories
        tvMealsLoggedValue.text = "$count/3"
        tvCaloriesValue.text = "$totalCals/3000"

        // Meal Summary card: show food names, or "..." if not logged yet
        tvBreakfastSummary.text = MealStore.getMeal("breakfast")?.foodName ?: "..."
        tvLunchSummary.text = MealStore.getMeal("lunch")?.foodName ?: "..."
        tvDinnerSummary.text = MealStore.getMeal("dinner")?.foodName ?: "..."
    }
}
